rosbag2_cpp/include/rosbag2_cpp/reader_interfaces/base_reader_interface.hpp

```diff
@@ -45,6 +46,8 @@ class ROSBAG2_CPP_PUBLIC BaseReaderInterface
 
   virtual void close() = 0;
 
+  virtual void set_read_order(rosbag2_storage::ReadOrder) = 0;
```

Comment:
@emersonknapp I noticed that you ignored my suggestions by passing `ReaderOrder` argument by const reference without any explanations.
`ReadOrder` became if not class, but at least a struct and not an integral type like simple enum or integer value any more. It would make sense to use const reference to passing it as argument in setter methods. At least I don't see a reason to not do this.
I understand that such changes in interfaces is tedious and might looks like a unnecessary head ache. I did spent some time and prepared proposals for changes in all places where it's needed. I've even tried to compile it locally after changes.

Wouldn't you mind to apply my suggestions in batch commit? I believe it will not take too much your time to click buttons on web interface for adding changes in one batch commit.

Reply:
I was/am of the opinion that pass-by-value should be allowed sometimes because the value is small. However, I see that I did not say that explicitly, I was too busy talking about `struct`. I'm ready to throw away this entire interface in favor of `set_read_order(bool)`, since I have no specific plan to implement File-order reading in `sqlite3` and maybe it doesn't even make sense anyways as a use case.